### PR TITLE
Coremark development 19.05

### PR DESCRIPTION
**Pull request by reclusejack**
This PR includes several items listed below.
1. latest coremark repo
2. freedom-metal with chicken bit patch
3. bsps with chicken bits and core-series tag
4. compile options for optimization and for selecting proper linker scripts
5. readme update

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -20,6 +20,15 @@ endif
 PROGRAM ?= hello
 TARGET ?= sifive-hifive1
 
+# Coremark require PORT_DIR set for different OS, freedom-metal for us!
+ifeq ($(PORT_DIR),)
+#$(error PORT_DIR not set for $(PROGRAM))
+ifeq ($(PROGRAM),coremark)
+PORT_DIR = freedom-metal
+#$(error PORT_DIR set to $(PORT_DIR), for $(PROGRAM))
```
Comment (nategraff-sifive) [inline Makefile]: These lines are still commented out.

Comment (reclusejack) [reply Makefile]: Updated the change, that should be for debugging purpose. We don't need it now.

**Comment by reclusejack** [comment]
> Where did the metal.ramrodata.coremark.lds files get generated? I don't see a corresponding PR to freedom-devicetree-tools.

Got it, I will update the freedom-devicetree-tools and update this PR.

File: Makefile (hunk 1 above)
Comment (reclusejack) [reply Makefile]: I've sent a PR to freedom-devicetree-tools.

File: Makefile [hunk 2]
```diff
@@ -20,6 +20,13 @@ endif
 PROGRAM ?= hello
 TARGET ?= sifive-hifive1
 
+# Coremark require PORT_DIR set for different OS, freedom-metal for us!
+ifeq ($(PORT_DIR),)
+ifeq ($(PROGRAM),coremark)
```
Comment (bsousi5) [inline Makefile]: Per comment, PORT_DIR is a coremark thing, so I think it make more sense to flip the check.

Comment (reclusejack) [reply Makefile]: patch updated.

**Comment by reclusejack** [comment]
@cgsfv @jim-wilson 
We want to add the compile option: -mtune=sifive-7-series to get the best coremark score for 7-series on 19.05. Do we have this option on freedom-studio 19.05 release?

**Comment by jim-wilson** [comment]
-mtune=sifive-7-series was added to the 19.02 (19.03?) release, and will also be present in the 19.05 release.

**Comment by reclusejack** [comment]
@jim-wilson Thanks for the information ~

**Comment by cgsfv** [comment]
No, the -mtune=sifive-7-series was first added in a /work/dev-tools/preview/riscv64-unknown-elf-gcc-8.2.0-2019.02.0-sifive-7-integration-x86_64* build in the beginning of April. Currently it is also available in /work/dev-tools/release/riscv64-unknown-elf-gcc-8.2.0-2019.05.0-x86_64*. None of these build a official yet, and are thus not available through our website.

File: .gitmodules [hunk 3]
```diff
@@ -52,3 +52,6 @@
 [submodule "software/example-user-syscall"]
 	path = software/example-user-syscall
 	url = https://github.com/sifive/example-user-syscall.git
+[submodule "software/coremark"]
+	path = software/coremark
+	url = https://github.com/sifive/coremark.git
```
Comment (nategraff-sifive) [inline .gitmodules]: This should probably point at sifive/benchmark-coremark.git

File: scripts/standalone.mk [hunk 4]
```diff
@@ -43,6 +43,12 @@ LINK_TARGET = ramrodata
 endif
 endif
 
+ifeq ($(PROGRAM),coremark)
+ifeq ($(LINK_TARGET),)
+LINK_TARGET = ramrodata
```
Comment (nategraff-sifive) [inline scripts/standalone.mk]: Does this fit in the 16k of ram on the freedom-e310-arty and sifive-hifive1*? If it doesn't we need to disable this option for those targets.

File: bsp/coreip-e20-arty/metal-inline.h [hunk 5]
```diff
@@ -1,7 +1,6 @@
 /* Copyright 2019 SiFive, Inc */
 /* SPDX-License-Identifier: Apache-2.0 */
 /* ----------------------------------- */
-/* [XXXXX] 28-05-2019 10-05-52        */
 /* ----------------------------------- */
```
Comment (nategraff-sifive) [inline bsp/coreip-e20-arty/metal-inline.h]: If you rebase on development-19.05 it'll clean up the diff for all of these changes.

File: .gitmodules (hunk 3 above)
Comment (reclusejack) [reply .gitmodules]: Updated.

File: bsp/coreip-e20-arty/metal-inline.h (hunk 5 above)
Comment (reclusejack) [reply bsp/coreip-e20-arty/metal-inline.h]: Yes. I've rebased the develop-19.05 branch. Thanks :)

File: scripts/standalone.mk (hunk 4 above)
Comment (reclusejack) [reply scripts/standalone.mk]: Thanks for reminding. Yes, I encountered the itim size too small problem with CONFIGURATION=debug. To get the best coremark score, users should set CONFIGURATION to  "release". So, I've updated the lds selection~

Comment (nategraff-sifive) [reply scripts/standalone.mk]: That technically fixes the problem, but what I actually mean is this: https://github.com/sifive/freedom-e-sdk/blob/development-19.05/scripts/standalone.mk#L44

Comment (reclusejack) [reply scripts/standalone.mk]: For freedom-e310 and hifive1 series, it fits in the 16k of ram.
  text   data   bss     dec    hex   filename
28346 6784 5330  40460 9e0c ./coremark.elf

Comment (nategraff-sifive) [reply scripts/standalone.mk]: Awesome, perfect.

**Pull request merged**